src/bors/handlers/review.rs

```diff
@@ -0,0 +1,261 @@
+    notify_of_approval(&repo_state, pr, approver.as_str()).await
+}
+
+/// Unapprove a pull request.
+/// Pull request's author can also unapprove the pull request.
+pub(super) async fn command_unapprove(
+    repo_state: Arc<RepositoryState>,
+    db: Arc<PgDbClient>,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> anyhow::Result<()> {
+    tracing::info!("Unapproving PR {}", pr.number);
+    if insufficient_unapprove_permission(repo_state.clone(), pr, author) {
+        deny_unapprove_request(&repo_state, pr, author).await?;
+        return Ok(());
+    };
+    db.unapprove(repo_state.repository(), pr.number).await?;
+    handle_label_trigger(&repo_state, pr.number, LabelTrigger::Unapproved).await?;
+    notify_of_unapproval(&repo_state, pr).await
+}
+
+fn insufficient_approve_permission(repo: Arc<RepositoryState>, author: &GithubUser) -> bool {
+    !repo
+        .permissions
+        .load()
+        .has_permission(author.id, PermissionType::Review)
+}
+
+async fn deny_approve_request(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> anyhow::Result<()> {
+    tracing::warn!(
+        "Permission denied for approve command by {}",
+        author.username
+    );
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "@{}: :key: Insufficient privileges: not in review users",
+                author.username
+            )),
+        )
+        .await
+}
+
+async fn notify_of_approval(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    approver: &str,
+) -> anyhow::Result<()> {
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "Commit {} has been approved by `{}`",
+                pr.head.sha, approver
+            )),
+        )
+        .await
+}
+
+fn insufficient_unapprove_permission(
+    repo: Arc<RepositoryState>,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> bool {
+    author.id != pr.author.id
+        && !repo
+            .permissions
+            .load()
+            .has_permission(author.id, PermissionType::Review)
+}
+
+async fn deny_unapprove_request(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> anyhow::Result<()> {
+    tracing::warn!(
+        "Permission denied for unapprove command by {}",
+        author.username
+    );
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "@{}: :key: Insufficient privileges: not in review users",
+                author.username
+            )),
+        )
+        .await
+}
+
+async fn notify_of_unapproval(repo: &RepositoryState, pr: &PullRequest) -> anyhow::Result<()> {
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!("Commit {} has been unapproved", pr.head.sha)),
+        )
+        .await
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
These tests only check the labels, we should also look inside the DB after the tests to see if the status has been changed there. You could introduce a helper that takes the `World` (or `Borstester`) and checks approved/unapproved status in the DB and also in the labels.